examples/core/rgcn/hetero_rgcn.py

```diff
@@ -0,0 +1,654 @@
+            # in biased outcomes or sub-optimal performance if the model was
+            # previously stuck in a poor local minimum.
+            ##################################################################
+            pass
+
+        # `itertools.chain()` is a function in Python's itertools module.
+        # It is used to flatten a list of iterables, making them act as
+        # one big iterable.
+        # In this context, the following code is used to create a single
+        # iterable over the parameters of both the model and the embed_layer,
+        # which is passed to the optimizer. The optimizer then updates all
+        # these parameters during the training process.
+        all_params = itertools.chain(
+            model.parameters(), embed_layer.parameters()
+        )
+        optimizer = th.optim.Adam(all_params, lr=0.01)
+
+        # If CPU affinity is supported and the device is CPU, execute training
+        # with CPU affinity enabled.
+        if args.num_workers != 0 and device == "cpu" and is_support_affinity():
+            # `expected_max`` is the number of physical cores on your machine.
+            # The `logical` parameter, when set to False, ensures that the count
+            # returned is the number of physical cores instead of logical cores
+            # (which could be higher due to technologies like Hyper-Threading).
+            expected_max = int(psutil.cpu_count(logical=False))
+            if args.num_workers >= expected_max:
+                print(
+                    "[ERROR] You specified num_workers are larger than physical"
+                    f"cores, please set any number less than {expected_max}",
+                    file=sys.stderr,
+                )
+            with train_loader.enable_cpu_affinity():
+                logger = train(
+                    g,
+                    model,
+                    embed_layer,
+                    optimizer,
+                    train_loader,
+                    split_idx,
+                    labels,
+                    logger,
+                    device,
+                    run,
+                )
+        else:
+            # If CPU affinity is not supported or the device is GPU, execute
+            # normal training.
+            logger = train(
+                g,
+                model,
+                embed_layer,
+                optimizer,
+                train_loader,
+                split_idx,
+                labels,
+                logger,
+                device,
+                run,
+            )
+        logger.print_statistics(run)
+
+    print("Final performance: ")
+    logger.print_statistics()
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="RGCN")
+    parser.add_argument("--runs", type=int, default=10)
+    parser.add_argument("--num_workers", type=int, default=0)
+
+    args = parser.parse_args()
+
+    main(args)
```

Comment:
why do you need to wrap a main function? why the parse is not in the main function?